python/arcticdb/version_store/library.py

```diff
@@ -1120,9 +1120,23 @@ def finalize_staged_data(
         prune_previous_versions: bool = False,
         metadata: Any = None,
         validate_index = True,
+        drop_staged_data_on_failure: bool = False
     ) -> VersionedItem:
         """
-        Finalizes staged data, making it available for reads.
+        Finalizes staged data, making it available for reads. All segments must be ordered and non-overlapping.
+        ``finalize_staged_data`` is less time consuming than ``sort_and_finalize_staged_data``.
+
+        If ``mode`` is ``StagedDataFinalizeMethod.APPEND`` the index of the first row of the sorted block must be equal to or greater
+        than the index of the last row in the existing data.
+
+        If ``Static Schema`` is used all staged block must have matching schema (same column names, same dtype, same column ordering).
+        For more information about schema options see documentation for ``arcticdb.LibraryOptions.dynamic_schema``
+
+        If the symbol does not exist both ``StagedDataFinalizeMethod.APPEND`` and ``StagedDataFinalizeMethod.WRITE`` create it.
+
+        Calling ``finalize_staged_data`` without having staged data for the symbol is not allowed.
```

Comment:
"is not allowed" -> "will throw a blah exception".
Link to the method that checks if there is staged data for a symbol too